fhir-server/src/test/java/au/csiro/pathling/fhirpath/function/WhereFunctionTest.java

```diff
@@ -329,4 +296,45 @@ public void throwsErrorIfArgumentNotSingular() {
         error.getMessage());
   }
 
+  @Test
+  public void throwsErrorIfInputIsLiteral() {
+    final ElementPath argument = new ElementPathBuilder()
+        .expression("$this.gender")
+        .fhirType(FHIRDefinedType.STRING)
+        .build();
+    final BooleanLiteralPath input = BooleanLiteralPath
+        .fromString("true", argument);
+
+    final ParserContext parserContext = new ParserContextBuilder().build();
+    final NamedFunctionInput whereInput = new NamedFunctionInput(parserContext, input,
+        Collections.singletonList(argument));
+
+    final NamedFunction whereFunction = NamedFunction.getInstance("where");
+    final InvalidUserInputError error = assertThrows(
+        InvalidUserInputError.class,
+        () -> whereFunction.invoke(whereInput));
+    assertEquals(
+        "Input to where function cannot be a literal: true",
+        error.getMessage());
+  }
+
+  @Test
+  public void throwsErrorIfArgumentIsLiteral() {
```

Comment:
I am not sure if it would be useful but speaking why:
```
xxxx.where(true) 
```
should be illegal?

Reply:
It's not illegal per the spec - but it adds quite a bit of complexity if we want to support it, and it's not very useful.